app/src/main/java/org/wikipedia/search/SearchResultsFragment.kt

```diff
@@ -0,0 +1,532 @@
+            }
+
+            // highlight search term within the text
+            boldenKeywordText(pageTitleText, pageTitle.displayText, currentSearchTerm)
+            searchResultItemImage.visibility = if (pageTitle.thumbUrl == null) if (type === SearchResult.SearchResultType.SEARCH) View.GONE else View.INVISIBLE else View.VISIBLE
+            loadImageWithRoundedCorners(searchResultItemImage, pageTitle.thumbUrl)
+
+            // ...and lastly, if we've scrolled to the last item in the list, then
+            // continue searching!
+            if (position == totalResults.size - 1 && WikipediaApp.getInstance().isOnline) {
+                if (lastFullTextResults == null) {
+                    // the first full text search
+                    doFullTextSearch(currentSearchTerm, null, false)
+                } else if (!lastFullTextResults!!.continuation.isNullOrEmpty()) {
+                    // subsequent full text searches
+                    doFullTextSearch(currentSearchTerm, lastFullTextResults!!.continuation, false)
+                }
+            }
+            view.isLongClickable = true
+            view.setOnClickListener {
+                if (position < totalResults.size) {
+                    callback()?.navigateToTitle(totalResults[position].pageTitle, false, position)
+                }
+            }
+            view.setOnCreateContextMenuListener(LongPressHandler(view,
+                    pageTitle, HistoryEntry.SOURCE_SEARCH, SearchResultsFragmentLongPressHandler(position)))
+        }
+    }
+
+    private fun cache(resultList: List<SearchResult>, searchTerm: String) {
+        val cacheKey = "$searchLanguageCode-$searchTerm"
+        searchResultsCache[cacheKey]?.let {
+            it.addAll(resultList)
+            searchResultsCache.put(cacheKey, it)
+        }
+    }
+
+    private fun log(resultList: List<SearchResult>, startTime: Long) {
+        // To ease data analysis and better make the funnel track with user behaviour,
+        // only transmit search results events if there are a nonzero number of results
+        if (resultList.isNotEmpty()) {
+            // noinspection ConstantConditions
+            callback()?.getFunnel()?.searchResults(true, resultList.size, displayTime(startTime), searchLanguageCode)
+        }
+    }
+
+    private fun logError(fullText: Boolean, startTime: Long) {
+        // noinspection ConstantConditions
+        callback()?.getFunnel()?.searchError(fullText, displayTime(startTime), searchLanguageCode)
+    }
+
+    private fun displayTime(startTime: Long): Int {
+        return TimeUnit.NANOSECONDS.toMillis(System.nanoTime() - startTime).toInt()
+    }
+
+    private fun callback(): Callback? {
+        return getCallback(this, Callback::class.java)
+    }
+
+    private val searchLanguageCode: String
+        get() = (parentFragment as SearchFragment).searchLanguageCode
```

Comment:
Should we use `requireParentFragment()` on this one and also add "`isAdded` to prevent fragment not being attached?